lib/resonate.ts

```diff
@@ -578,8 +597,22 @@ class ResonateContext implements Context {
         if (isPendingPromise(promise)) {
           throw new Error("Invalid state");
         } else if (isResolvedPromise(promise)) {
+          if (
+            this.opts.test !== undefined &&
```

Comment:
Let's simplify this a bit, if we change `opts.test` to a number and default it to 0 we should be able to do this.
```
const fail = Math.random() < this.opts.test;
const branch =  Math.floor(Math.random() * 3);

if(fail && branch === 0) {
  ...
}
```